services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala

```diff
@@ -1,18 +1,38 @@
 package cromwell.pipeline.service
 
 import cromwell.pipeline.datastorage.dao.repository.DocumentRepository
+import cromwell.pipeline.datastorage.dto.project.configuration.{
+  ProjectConfigurationRequest,
+  ProjectConfigurationResponse
+}
 import cromwell.pipeline.datastorage.dto.{ ProjectConfiguration, ProjectId }
 
 import scala.concurrent.{ ExecutionContext, Future }
 
 class ProjectConfigurationService(repository: DocumentRepository)(implicit ec: ExecutionContext) {
   import ProjectConfiguration._
 
-  def addConfiguration(projectConfiguration: ProjectConfiguration): Future[String] =
+  def addConfiguration(configuration: ProjectConfigurationRequest): Future[String] =
     repository
-      .updateOne(toDocument(projectConfiguration), "projectId", projectConfiguration.projectId.value)
+      .updateOne(
+        toDocument(ProjectConfiguration(configuration.projectId, configuration.projectFileConfigurations)),
+        "projectId",
+        configuration.projectId.value
+      )
       .map(_.toString)
 
-  def getById(projectId: ProjectId): Future[Option[ProjectConfiguration]] =
-    repository.getByParam("projectId", projectId.value).map(_.headOption.map(fromDocument))
+  def getById(projectId: ProjectId): Future[Option[ProjectConfigurationResponse]] =
+    repository.getByParam("projectId", projectId.value).map(_.headOption.map(fromDocument)).map {
+      case Some(conf) if conf.isActive =>
+        Some(ProjectConfigurationResponse(conf.projectId, conf.projectFileConfigurations))
+      case _ => None
+    }
+
+  def deactivateConfiguration(projectId: ProjectId): Future[String] = {
+    repository.getByParam("projectId", projectId.value).map(_.headOption.map(fromDocument)).flatMap {
+      case Some(conf) if conf.isActive =>
+        repository.updateOneField("projectId", projectId.value, "isActive", false).map(_.toString)
```

Comment:
Why not use`updateOne` method here? I don't think we want to keep creating methods like `updateTwoFields`, `updateThreeFields`, ets.

Reply:
I thought update one field operation is faster then update whole document

Reply:
Maybe you are right, but I think for now the readability of the code is more important, than its performance. @ilyaonishenko what do you think?

Reply:
Forgot that Ilya is temporarily unavailable. @mnesvit what do you think? :)

Reply:
Discussed it with @mnesvit and @ilyaonishenko at a meeting, decided that updateOneField method has right to exist ;)